include/seqan/sequence/string_packed.h

```diff
@@ -579,10 +583,74 @@ resize(
         clear(host(me));
         return 0;
     }
-    TStringSize max_length = (resize(host(me), TTraits::toHostLength(new_length) + 1, tag) - 1) * TTraits::VALUES_PER_HOST_VALUE;
-    if ((TStringSize)new_length > max_length)
-        new_length = max_length;
-    return front(host(me)).i = new_length;
+
+    TStringSize _new_length = new_length;
+    TStringSize max_length = (resize(host(me), TTraits::toHostLength(_new_length) + 1, tag) - 1) * TTraits::VALUES_PER_HOST_VALUE;
+    if (_new_length > max_length)
+        _new_length = max_length;
+    front(host(me)).i = _new_length;
+
+    return _new_length;
+}
+
+template <typename TValue, typename THostspec, typename TSize, typename TExpand, typename TValue2>
+inline typename Size< String<TValue, Packed<THostspec> > >::Type
+resize(String<TValue, Packed<THostspec> > & me,
+       TSize new_length,
+       TValue2 const & newValue,
+       Tag<TExpand> tag)
+{
+    typedef String<TValue, Packed<THostspec> > TString;
+    typedef PackedTraits_<TString> TTraits;
+    typedef typename Size<TString>::Type TStringSize;
+
+    TStringSize _new_length = new_length;
+    TStringSize old_length = length(me);
+
+    if (_new_length < length(me))
+        return resize(me, _new_length, tag);
+
+    // create WORD
+    String<TValue, Packed<THostspec> > tmp;
+    resize(tmp, static_cast<unsigned>(TTraits::VALUES_PER_HOST_VALUE), Exact());
+    for (unsigned i = 0; i < TTraits::VALUES_PER_HOST_VALUE; ++i)
+        tmp[i] = newValue;
+
+    // fill up host string word-wise
+    TStringSize max_length = (resize(host(me),
+                                     TTraits::toHostLength(_new_length) + 1,
+                                     getValue(host(tmp), 1),
+                                     tag) - 1)
+                             * TTraits::VALUES_PER_HOST_VALUE;
+
+    // set values on formerly last word, because not affected by above
+    TStringSize trailing_values = TTraits::VALUES_PER_HOST_VALUE - (old_length % TTraits::VALUES_PER_HOST_VALUE);
+    for (TStringSize i = 0; i < trailing_values; ++i)
+        me[old_length + i] = newValue;
```

Comment:
Again just mask the word at the corresponding positions in the host value. 
Using subscript operator creates a certain overhead.